L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc

```diff
@@ -0,0 +1,327 @@
+/*!  \brief   Implementation of methods of TTClusterBuilder.h
+ *  \details Here, in the source file, the methods which do depend
+ *           on the specific type <T> that can fit the template.
+ *
+ * \author Andrew W. Rose
+ * \author Nicola Pozzobon
+ * \author Ivan Reid
+ * \date 2013, Jul 18
+ *
+ */
+
+#include "L1Trigger/TrackTrigger/plugins/TTStubBuilder.h"
+
+/// Implement the producer
+template< >
+void TTStubBuilder< Ref_Phase2TrackerDigi_ >::produce( edm::Event& iEvent, const edm::EventSetup& iSetup )
+{
+  //Retrieve tracker topology from geometry                                                                                                              
+  edm::ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  const TrackerTopology* const tTopo = tTopoHandle.product();
+  edm::ESHandle< TrackerGeometry > tGeomHandle;
+  iSetup.get< TrackerDigiGeometryRecord >().get( tGeomHandle );
+  const TrackerGeometry* const theTrackerGeom = tGeomHandle.product();
+	
+  /// Prepare output
+  std::auto_ptr< edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > > TTClusterDSVForOutput( new edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > );
```

Comment:
You could save yourself some typing by doing
```C++
auto ttClusterDSVForOutput = std::make_unique<edmNew::DetSetVector<TTCluster<Ref_Phase2TrackerDigi_>>>();
```

using `std::make_unique` requires that you only type the typename once plus it properly handles cases where an exception happens when the object is being made so memory gets cleaned up.

Also note, it is CMS policy that variable names begin with a lower case letter.